src/Data/AffineSpace.hs

```diff
@@ -122,3 +140,58 @@ instance (AffineSpace p) => AffineSpace (a -> p) where
   type Diff (a -> p) = a -> Diff p
   (.-.)              = liftA2 (.-.)
   (.+^)              = liftA2 (.+^)
+
+
+
+newtype GenericDiff p = GenericDiff (Diff (Gnrx.Rep p ()))
+       deriving (Generic)
+
+instance AdditiveGroup (Diff (Gnrx.Rep p ())) => AdditiveGroup (GenericDiff p)
+instance VectorSpace (Diff (Gnrx.Rep p ())) => VectorSpace (GenericDiff p)
+instance InnerSpace (Diff (Gnrx.Rep p ())) => InnerSpace (GenericDiff p)
+instance HasBasis (Diff (Gnrx.Rep p ())) => HasBasis (GenericDiff p)
+
+data AffineDiffProductSpace f g p = AffineDiffProductSpace
```

Comment:
Do we need a new `data` type? Any why strict fields? How about using `f :*: g` instead?